### DISCUSSION TITLE
Field split for scalar variables

### DISCUSSION THREAD

**Discussion by ttruster** [idaholab/moose#24356] [Q&A General] [answered]
Regarding the field split preconditioner:

I noticed in `Split.C` that there is a guard against scalar variables:
`if (!_fe_problem.hasVariable(var))`
However, lower in the code (`PetscDMMoose.C`) it actually does allow for scalar variables. When I commented the line above and ran a test case combining a regular and scalar variable with field splitting, the iterative solvers ran fine.

Is this guard intentional or an oversight? If nothing else prevents changing it to
`if (!_fe_problem.hasVariable(var) && !_fe_problem.hasScalarVariable(var))`
then I can do that, add a test input file, and also fix a typo I had in one of my doc files in a previous pull request.

**Comment by GiudGiud** [comment]
Hello

Which syntax are you using for the field split? There's two
I think it got enabled recently for the `[Preconditioning] type=FSP` syntax, that may be why it s disallowed in your case. 

Guillaume

**Reply by ttruster** [reply]
I am using the `[Preconditioning]  active = 'FSP'` syntax, not the one that Contact Split is based off of. However, I'm using the source code from 4/7/2023. But when I look in the current github source, the `Split.C` file looks the same. I can try the [Splits] syntax, but when I skim the source it looks like both syntaxes will pass through that same source file.

Let me know if you want me to attach the input file.

**Reply by lindsayad** [reply, marked as answer]
I would open a PR with a test case so we can review. Sounds like a possible oversight
